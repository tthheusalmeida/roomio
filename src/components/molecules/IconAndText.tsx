"use client";

import { ComponentType } from "react";

interface IconConfig {
  component?: ComponentType<{ size?: number; className?: string }>;
  size?: number;
}

interface IconAndTextProps {
  icon?: IconConfig;
  text?: string;
  iconOnStart?: boolean;
}

export function IconAndText({
  icon,
  text,
  iconOnStart = true,
}: IconAndTextProps) {
  const { component: IconComponent, size: IconSize, ...iconProps } = icon ?? {};

  const iconElement = IconComponent ? (
    <IconComponent size={IconSize ?? 20} {...iconProps} />
  ) : null;

  return (
    <div className="flex justify-center gap-4 items-center">
      {iconOnStart && iconElement}
      {text && <span className="text-sm font-normal">{text}</span>}
      {!iconOnStart && iconElement}
    </div>
  );
}
