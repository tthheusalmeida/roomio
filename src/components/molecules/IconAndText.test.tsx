import { render, screen } from "@testing-library/react";
import { IconAndText } from "./IconAndText";

// Mock de um ícone genérico
const MockIcon = ({ size }: { size?: number }) => (
  <span data-testid="icon" style={{ fontSize: size }}>{`Icon ${size}`}</span>
);

describe("IconAndText component", () => {
  it("renders text correctly", () => {
    const { asFragment } = render(<IconAndText text="Hello" />);
    const textElement = screen.getByText("Hello");
    expect(textElement).toBeInTheDocument();
    expect(textElement.tagName).toBe("SPAN");
    expect(textElement).toHaveClass("text-sm font-normal");

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders icon when provided", () => {
    render(<IconAndText icon={{ component: MockIcon, size: 32 }} />);
    const icon = screen.getByTestId("icon");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveStyle({ fontSize: "32px" });
  });

  it("renders icon before text when iconOnStart is true", () => {
    render(
      <IconAndText icon={{ component: MockIcon }} text="Hello" iconOnStart />
    );
    const container = screen.getByText("Hello").parentElement!;
    expect(container.firstChild).toHaveAttribute("data-testid", "icon");
    expect(container.lastChild).toHaveTextContent("Hello");
  });

  it("renders icon after text when iconOnStart is false", () => {
    render(
      <IconAndText
        icon={{ component: MockIcon }}
        text="Hello"
        iconOnStart={false}
      />
    );
    const container = screen.getByText("Hello").parentElement!;
    expect(container.firstChild).toHaveTextContent("Hello");
    expect(container.lastChild).toHaveAttribute("data-testid", "icon");
  });

  it("matches snapshot with icon and text", () => {
    const { asFragment } = render(
      <IconAndText icon={{ component: MockIcon }} text="Hello" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
