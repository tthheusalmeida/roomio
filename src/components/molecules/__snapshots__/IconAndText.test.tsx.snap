// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`IconAndText component matches snapshot with icon and text 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center gap-4 items-center"
  >
    <span
      data-testid="icon"
      style="font-size: 24px;"
    >
      Icon 24
    </span>
    <span
      class="text-sm font-normal"
    >
      Hello
    </span>
  </div>
</DocumentFragment>
`;

exports[`IconAndText component renders text correctly 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center gap-4 items-center"
  >
    <span
      class="text-sm font-normal"
    >
      Hello
    </span>
  </div>
</DocumentFragment>
`;
